### PR TITLE
Add null filter and refactor config

### PR DESCRIPTION
**Pull request by domoritz**

### CODE REVIEW

File: src/compiler/compiler.ts [hunk 1]
```diff
@@ -32,15 +32,15 @@ export function compile(spec, theme?) {
       padding: 'auto'
     },
     ['viewport', 'background'].reduce(function(topLevelConfig, property) {
-      const value = model.config(property);
+      const value = model.config()[property];
```
Comment (kanitw) [inline src/compiler/compiler.ts]: should `model.config()` and `model.sceneConfig()` be cached in consts above?  

(No worry if you prefer this)

Comment (domoritz) [inline src/compiler/compiler.ts]: Nah, it's only used in one line.

File: src/compiler/data.ts [hunk 2]
```diff
@@ -175,9 +182,10 @@ export namespace source {
    * @return An array that might contain a filter transform for filtering null value based on filterNul config
    */
   export function nullFilterTransform(model: Model) {
-    const filterNull = model.config('filterNull');
+    const filterNull = model.config().filterNull;
     const filteredFields = keys(model.reduce(function(aggregator, fieldDef: FieldDef) {
-      if (fieldDef.field && fieldDef.field !== '*' && filterNull[fieldDef.type]) {
+      if (filterNull === true ||
+        filterNull !== false && fieldDef.field && fieldDef.field !== '*' && DEFAULT_NULL_FILTERS[fieldDef.type]) {
```
Comment (kanitw) [inline src/compiler/data.ts]: - `filterNull === true` => `filterNull` ?
- `filterNull !== false` => `filterNull === undefined` – clearer
- Please add parentheses for readability!

File: src/schema/config.schema.ts [hunk 3]
```diff
@@ -57,13 +57,9 @@ export const config = {
...
     // TODO(#597) revise this config
     filterNull: {
-      type: 'object',
-      properties: {
-        nominal: {type:'boolean', default: false},
-        ordinal: {type:'boolean', default: false},
-        quantitative: {type:'boolean', default: true},
-        temporal: {type:'boolean', default: true}
-      }
+      type: 'boolean',
```
Comment (kanitw) [inline src/schema/config.schema.ts]: Please copy this to `config.md`

File: src/schema/config.schema.ts [hunk 4]
```diff
@@ -57,13 +57,9 @@ export const config = {
     // filter null
     // TODO(#597) revise this config
```
Comment (kanitw) [inline src/schema/config.schema.ts]: remove this line.

File: src/compiler/data.ts (hunk 2 above)
Comment (domoritz) [inline src/compiler/data.ts]: Hmm, I wonder whether 'auto' should also work. That's why I wrote it the way it is. But we should be consistent.

**Comment by kanitw** [comment]
Please also modify VLUI’s `toggleNull` button in `vlplot.js` based on this change.

File: src/compiler/compiler.ts (hunk 1 above)
Comment (kanitw) [inline src/compiler/compiler.ts]: Fine

File: src/compiler/data.ts (hunk 2 above)
Comment (kanitw) [inline src/compiler/data.ts]: Nope.  This is boolean, `'auto'` is not boolean.

Comment (domoritz) [inline src/compiler/data.ts]: This is javascript. Anything can happen :-P

Comment (kanitw) [inline src/compiler/data.ts]: It should pass the schema validator from the beginning once we have it in place.

**Comment by kanitw** [comment]
LGTM, feel free merge once you solve all issue and add VLUI fixes

File: src/compiler/Model.ts [hunk 5]
```diff
@@ -232,17 +232,17 @@ export class Model {
...
+    return this._spec.config;
   }
 
   /**
-   * @return Cell config value from the spec, or a default value if unspecified.
+   * Get the spec configuration for cells.
    */
-  public cellConfig(name: string) {
-    return this._spec.config.cell[name];
+  public cellConfig() {
```
Comment (kanitw) [inline src/compiler/Model.ts]: Actually, should we just remove `scene|cellConfig()` and use `config().scene|cell` ?

Comment (domoritz) [inline src/compiler/Model.ts]: Thought about it as well. Yes.

**Pull request merged**